rpcs3/Emu/RSX/Program/GLSLCommon.cpp

```diff
@@ -547,6 +516,15 @@ namespace glsl
 		OS << "#define _test_bit(x, y) (_get_bits(x, y, 1) != 0)\n";
 		OS << "#define _rand(seed) fract(sin(dot(seed.xy, vec2(12.9898f, 78.233f))) * 43758.5453f)\n\n";
 
+		if (props.low_precision_tests)
+		{
+			OS << "#define CMP_FIXUP(a) (sign(a) * 16. + a)\n\n";
```

Comment:
won't this add double precision to otherwise float calculations?
or is it not an issue and i misunderstood

Reply:
GPU code is different than CPU code.
16.0 is float just like 16. and 16.f
Doubles are special and have to be declared explicitly. They also don't have vector types.

Reply:
FTR double would be 16.lf